### PR TITLE
fix SHOW SERIES

### PR DESCRIPTION
**Pull request by dgnorton**

### CODE REVIEW

File: httpd/handler_test.go [hunk 1]
```diff
@@ -796,17 +799,120 @@ func TestHandler_serveShowSeries(t *testing.T) {
...
+								Values: [][]interface{}{
+									str2iface([]string{"server02", "useast"}),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		// SHOW SERIES ... LIMIT
```
Comment (otoolep) [inline httpd/handler_test.go]: On purpose?

File: server.go [hunk 2]
```diff
@@ -1767,7 +1768,11 @@ func (s *Server) executeShowSeriesStatement(stmt *influxql.ShowSeriesStatement,
 	if stmt.Source != nil {
 		// TODO: handle multiple measurement sources
 		if m, ok := stmt.Source.(*influxql.Measurement); ok {
-			measurements = append(measurements, m.Name)
+			segments, err := influxql.SplitIdent(m.Name)
+			if err != nil {
+				return &Result{Err: err}
+			}
+			measurements = append(measurements, segments[2])
```
Comment (otoolep) [inline server.go]: What are segments 0 and 1?

File: httpd/handler_test.go (hunk 1 above)
Comment (dgnorton) [inline httpd/handler_test.go]: Yes, `LIMIT` still needs work

Comment (otoolep) [inline httpd/handler_test.go]: I mean, should the code just be removed?

File: server.go (hunk 2 above)
Comment (dgnorton) [inline server.go]: Ben added a change to normalize measurements (e.g. "cpu" becomes "db"."policy"."cpu") in the AST.  `influxql.SplitIdent` splits a normalized measurement into it's pieces.  The measurements map uses only the measurement name as the key, so it has to be split out here.

Comment (otoolep) [inline server.go]: Oh, the DB and RP?

File: server.go [hunk 3]
```diff
@@ -1814,13 +1829,13 @@ func (s *Server) executeShowSeriesStatement(stmt *influxql.ShowSeriesStatement,
 		// Loop through series IDs getting matching tag sets.
 		for _, id := range ids {
 			if s, ok := m.seriesByID[id]; ok {
-				values := make([]string, 0, len(r.Columns))
+				values := make([]interface{}, 0, len(r.Columns))
```
Comment (dgnorton) [inline server.go]: @otoolep  this is an important part of the fix because it fixes the final result format.

File: server.go [hunk 4]
```diff
@@ -1776,6 +1781,11 @@ func (s *Server) executeShowSeriesStatement(stmt *influxql.ShowSeriesStatement,
 		measurements = db.MeasurementNames()
 	}
 
+	// If OFFSET is past the end of the array, return empty results.
+	if stmt.Offset > len(measurements)-1 {
```
Comment (otoolep) [inline server.go]: Style: I would have just written `>=` but I'm really nitpicking here. :-)

**Comment by dgnorton** [comment]
@pauldix says "do it!"

File: server.go (hunk 2 above)
Comment (dgnorton) [inline server.go]: Yes

File: server.go (hunk 4 above)
Comment (dgnorton) [inline server.go]: I think I wrote that line around 6am?  :)

File: httpd/handler_test.go (hunk 1 above)
Comment (dgnorton) [inline httpd/handler_test.go]: Need to fix LIMIT then enable that test

**Pull request merged**